test/rpc/test_rpc_call.py

```diff
@@ -88,7 +88,7 @@ def test_call_raises_on_invalid_calldata(deploy_info):
             "request": {
                 "contract_address": pad_zero(contract_address),
                 "entry_point_selector": hex(get_selector_from_name("get_balance")),
-                "calldata": ["a", "b", "123"],
```

Comment:
Is this necessary?

Reply:
i think yes, because the checking needs to be as rigorous as possible

1. check if the datatype does not comply
2. check if the datatype is correct, but doesn't comply to the number of expected parameters supplied

what do you think about this?

@FabijanC

Reply:
Sure we need to test with invalid input, but I don't understand this two-item checklist that you provided - is that something you claim this particular test is doing? Or is it related to other tests as well (and how)? Perhaps pytest.mark.parametrize could be of use (already used in some rpc tests)

Reply:
oh yeah i think i should try to use `pytest.mark.parametrize`

thank you!